pkg/networkpolicy/controller.go

```diff
@@ -56,28 +51,9 @@ type Config struct {
 	QueueID  int
 }
 
-// detect if the system uses iptables legacy
-func iptablesLegacy() bool {
-	// only support IPv4 with iptables for simplicity
-	path, err := exec.LookPath("iptables")
-	if err != nil {
-		return false
-	}
-	cmd := exec.Command(path, "--version")
-	var out bytes.Buffer
-	cmd.Stdout = &out
-	err = cmd.Run()
-	if err != nil {
-		return false
-	}
-	if strings.Contains(out.String(), "legacy") {
-		return true
-	}
-	return false
-}
-
 // NewController returns a new *Controller.
 func NewController(client clientset.Interface,
+	nft knftables.Interface,
```

Comment:
Why pass `nft` into `NewController` but not `nfq`?

I feel like it makes more sense to have `Controller` create the knftables object itself. It knows what it needs; `main` doesn't.

Reply:
to mock it in the tests

Reply:
also nfq needs to register the hooks that depend on the controller